### ISSUE TITLE
KeyboardDragListener should drag with deltas instead of velocities

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/quadrilateral#33]
From https://github.com/phetsims/quadrilateral/issues/4

> - Since motion of vertices is velocity based, it is difficult to make reproducible cases. Lets change it to move a certain delta per key press (and as long as it is held down).

**Comment by jessegreenberg** [comment]
THis was done in the above commit. @terracoda and @BLFiedler can you please review this change? Anything you would like to behave differently? With this approach we have the following options with the drag handler:
- How far the object moves upon first press.
- How far the object moves each step in press and hold motion.
- How long to wait before continuing motion after the first press.
- How long to wait between each step during continuous motion.

**Comment by terracoda** [comment]
The timings and steps for the vertices and sides seems fine to me @jessegreenberg, but there are times when I cannot move a side.

When I move the sides as close together as they can go, I can't move the top and bottom sides left or right. And I can't move the right and left sides up or down.

<img width="1225" alt="image" src="https://user-images.githubusercontent.com/1902116/142485329-a7ae4b1c-5b6d-4a58-854e-7889275eed2e.png">

So, once in the position above I can't move top and bottom sides left and right, and I can't move the left and right sides up and down.

That seems weird.
And there are a couple other places where I noticed some sticky behaviour with the upper side and lower side.

Note, the current step size is smaller than the tolerance. I think these will eventually have to align.

**Comment by terracoda** [comment]
<img width="836" alt="image" src="https://user-images.githubusercontent.com/1902116/142487686-e9fe6760-4f09-42d5-b9b6-b22550c14e7d.png">

So in the case above I cannot move the top side left or right

**Comment by terracoda** [comment]
Assigning back to you @jessegreenberg to have a look at the sticky points.

**Comment by jessegreenberg** [comment]
Here is another case where I cannot drag the top side to the left:
![image](https://user-images.githubusercontent.com/6396244/145075081-d8b732b3-f0aa-4266-b9d0-8ff97c13983f.png)

In this example the drag areas are shown with ?dragAreas. I **can** drag the top side to the left with a mouse so there is a bug with keyboard in particular.

**Comment by jessegreenberg** [comment]
The model has been rewritten so that there is a model for the main quadrilateral and also a "scratch" quadrilateral so that we can check the positions of all vertices at once.

**Comment by jessegreenberg** [comment]
Alright, this is working quite a bit better and the proposal https://github.com/phetsims/quadrilateral/issues/33#issuecomment-988118665 is working out so far. It no longer gets stuck, but I still see that dragging is constrained to the computed drag areas when using both mouse and keyboard. Closing.
